src/main/scala/firrtl/passes/VerilogMemDelays.scala

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright (c) 2014 - 2016 The Regents of the University of
+California (Regents). All Rights Reserved.  Redistribution and use in
+source and binary forms, with or without modification, are permitted
+provided that the following conditions are met:
+   * Redistributions of source code must retain the above
+     copyright notice, this list of conditions and the following
+     two paragraphs of disclaimer.
+   * Redistributions in binary form must reproduce the above
+     copyright notice, this list of conditions and the following
+     two paragraphs of disclaimer in the documentation and/or other materials
+     provided with the distribution.
+   * Neither the name of the Regents nor the names of its contributors
+     may be used to endorse or promote products derived from this
+     software without specific prior written permission.
+IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT,
+SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS,
+ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF
+REGENTS HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE. THE SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF
+ANY, PROVIDED HEREUNDER IS PROVIDED "AS IS". REGENTS HAS NO OBLIGATION
+TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR
+MODIFICATIONS.
+*/
+
+package firrtl.passes
+
+import firrtl._
+import firrtl.ir._
+import firrtl.Utils._
+import firrtl.Mappers._
+import firrtl.PrimOps._
+
+/** This pass generates delay reigsters for memories for verilog */
+object VerilogMemDelays extends Pass {
+  def name = "Verilog Memory Delays"
+  val ug = UNKNOWNGENDER
+  type Netlist = collection.mutable.HashMap[String, Expression]
+  implicit def expToString(e: Expression) = e.serialize
+  private def NOT(e: Expression) = DoPrim(Not, Seq(e), Nil, BoolType)
+  private def AND(e1: Expression, e2: Expression) = DoPrim(And, Seq(e1, e2), Nil, BoolType)
+
+  def buildNetlist(netlist: Netlist)(s: Statement): Statement = {
+    s match {
+      case Connect(_, loc, expr) => kind(loc) match {
+        case MemKind(_) => netlist(loc) = expr
+        case _ =>
+      }
+      case _ =>
+    }
+    s map buildNetlist(netlist)
+  }
+
+  def memDelayStmt(netlist: Netlist, namespace: Namespace, repl: Netlist)(
+      s: Statement): Statement = s map memDelayStmt(netlist, namespace, repl) match {
```

Comment:
Prefer 1 argument per line for long method signatures

```scala
def memDelayStmt(
    netlist: Netlist,
    namespace: Namespace,
    repl: Netlist)
    (s: Statement): Statement = s map memDelayStmt(netlist, namespace, repl) match {
```

It might also be possible to fit the whole definition on one line if you put the `s map...` on the next line